### PR TITLE
cli: Make tinygo help experience more consistent with go help #3739

### PR DESCRIPTION
**Pull request by bjornpagen**
- Improve tinygo help command output by streamlining layout and command descriptions
- Reorganize commands and help topics alphabetically, moving info and targets to help topics
- Create a new 'flags' help topic that contains all the flag details
- Use 'text/tabwriter' for consistent formatting and structuring of the help output
- Remove "No command-line arguments supplied." to match go

Credit to @m-kru for layout

```
TinyGo is a Go compiler for small places.
version: 0.28.0-dev

Usage:

	tinygo <command> [arguments]

The commands are:

        build          compile packages and dependencies
        clean          empty cache directory (/Users/bjornpagen/Library/Caches/tinygo)
        env            list environment variables used during build
        flash          compile and flash to the device
        gdb            run/flash and immediately enter GDB
        help           print this help text
        list           run go list using the TinyGo root
        lldb           run/flash and immediately enter LLDB
        monitor        open communication port
        run            compile and run immediately
        test           test packages
        version        show version

Use "tinygo help <command>" for more information about a command.

Additional help topics:

        flags          general flags
        info           show info for specified target
        targets        list targets

Use "tinygo help <topic>" for more information about that topic.

For more details, see https://tinygo.org/docs/reference/usage/

```

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -1201,30 +1202,40 @@ func usage(command string) {
...
-			flag.PrintDefaults()
-		}
+		fmt.Fprintln(os.Stderr, "Usage:\n")
+		fmt.Fprintln(os.Stderr, "\ttinygo <command> [arguments]\n")
+		fmt.Fprintln(os.Stderr, "The commands are:\n")
+
+		tabw := tabwriter.NewWriter(os.Stderr, 0, 0, 8, ' ', 0)
+
+		fmt.Fprintln(tabw, "\tbuild\tcompile packages and dependencies")
+		fmt.Fprintln(tabw, "\tclean\tempty cache directory (/home/mkru/.cache/tinygo)")
```
Comment (m-kru) [inline main.go]: I wonder how my home path got in here.

Comment (bjornpagen) [reply main.go]: oops, fixed

File: main.go [hunk 2]
```diff
@@ -1607,6 +1617,11 @@ func main() {
 		if err != nil {
 			handleCompilerError(err)
 		}
+	case "flags":
+		if flag.Parsed() {
+			fmt.Fprintln(os.Stderr, "Flags:")
+			flag.PrintDefaults()
+		}
```
Comment (aykevl) [inline main.go:1624]: I'm not entirely sure about this. What about `tinygo help flags` (for lack of `tinygo help build` etc)?

You are also missing a bunch of flags that would be printed with `tinygo help`:

https://github.com/tinygo-org/tinygo/blob/ee90bdebfff426d8862906536ccbaf49d47c5409/main.go#L1434-L1447

**Review by aykevl — COMMENTED** [review]
Overall this looks like a good improvement.

**Comment by paralin** [comment]
Looks like a good improvement!

**Comment by deadprogram** [comment]
Now that https://github.com/tinygo-org/tinygo/pull/4467 was merged, seems like this PR would need a bunch of rework with that in mind. What do you think?

**Comment by deadprogram** [comment]
Closing since similar PR #4467 was merged. Thanks everyone!

**Pull request closed without merging**